lhotse/dataset/diarization.py

```diff
@@ -48,32 +50,52 @@ class DiarizationDataset(Dataset):
     """
 
     def __init__(
-            self,
-            cuts: CutSet,
-            min_speaker_dim: Optional[int] = None,
-            global_speaker_ids: bool = False,
+        self,
+        cuts: CutSet,
+        uem: Optional[SupervisionSet] = None,
+        min_speaker_dim: Optional[int] = None,
+        global_speaker_ids: bool = False,
     ) -> None:
         super().__init__()
         validate(cuts)
-        self.cuts = cuts
-        self.speakers = {spk: idx for idx, spk in enumerate(self.cuts.speakers)} if global_speaker_ids else None
+        self.cuts = (
+            cuts
+            if uem is None
+            else cuts.filter_supervisions(
+                lambda s: any(
+                    [
+                        s.start >= seg.start
+                        and s.start + s.duration <= seg.start + seg.duration
+                        for seg in uem
```

Comment:
If you expect `uem` to be large, it might makes sense to use an interval tree instead (see e.g. `index_supervisions`). But for small sets it should be quick anyway. You might want to consider `overlaps` or `overspans` from `lhotse.utils` for the comparison operations (the intent is more immediately readable)

Reply:
Cool! I wanted to use interval tree too but then I got lazy --- `index_supervisions` should be good.